### PR TITLE
qrencode: merge with libqrencode

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Closes: #53133

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @jameysharp @yegortimoshenko @adolfogc

### CODE REVIEW

**Review by lukateras — APPROVED** [review]

**Comment by jtojnar** [comment]
Apparently, the tests do not run in `make check`, we need to invoke `tests/test_basic.sh`.

**Comment by jameysharp** [comment]
Looks great to me, thanks!

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 1]
```diff
@@ -171349,12 +171349,12 @@ self: {
          base binary bytestring cryptonite directory haskell-qrencode
          jose-jwt JuicyPixels microlens process split vector
        ];
-       libraryPkgconfigDepends = [ libqrencode ];
+       libraryPkgconfigDepends = [ qrencode ];
        testHaskellDepends = [ base hspec ];
        description = "Library to generate images";
        license = stdenv.lib.licenses.bsd3;
        hydraPlatforms = stdenv.lib.platforms.none;
-     }) {inherit (pkgs) libqrencode;};
```
[the remark this answers is not in the document]
Comment (jtojnar) [reply pkgs/development/haskell-modules/hackage-packages.nix]: I think that cabal2nix finds the correct package through https://github.com/peti/distribution-nixpkgs

[the remark this answers is not in the document]
Comment (cdepillabout) [reply pkgs/development/haskell-modules/hackage-packages.nix]: Oh nice, thanks for the info @jtojnar, I didn't know about https://github.com/peti/distribution-nixpkgs!

**Pull request merged**